pontoon/api/tests/test_views.py

```diff
@@ -0,0 +1,165 @@
+import pytest
+
+from rest_framework.test import APIClient
+
+from pontoon.base.models.project import Project
+from pontoon.base.models.translation_memory import TranslationMemoryEntry
+from pontoon.terminology.models import Term, TermTranslation
+from pontoon.test.factories import LocaleFactory, ProjectFactory
+
+
+@pytest.fixture
+def locales():
+    return LocaleFactory.create_batch(3)
+
+
+@pytest.fixture()
+def regular_projects(locale_a):
+    return ProjectFactory.create_batch(3, visibility=Project.Visibility.PUBLIC) + list(
+        Project.objects.filter(slug__in=["terminology"])
+    )
+
+
+@pytest.fixture()
+def disabled_projects(locale_a):
+    return ProjectFactory.create_batch(3, disabled=True)
+
+
+@pytest.fixture()
+def system_projects(locale_a):
+    return ProjectFactory.create_batch(3, system_project=True) + list(
+        Project.objects.filter(slug__in=["tutorial"])
+    )
+
+
+@pytest.fixture
+def terms(locale_a):
+    term1 = Term.objects.create(
+        text="open",
+        part_of_speech="verb",
+        definition="Allow access",
+        usage="Open the door.",
+    )
+    term2 = Term.objects.create(
+        text="close",
+        part_of_speech="verb",
+        definition="Shut or block access",
+        usage="Close the door.",
+    )
+
+    TermTranslation.objects.create(term=term1, locale=locale_a, text="odpreti")
+    TermTranslation.objects.create(term=term2, locale=locale_a, text="zapreti")
+
+    return [term1, term2]
+
+
+@pytest.fixture
+def tm_entries(locale_a, locale_b, project_a, project_b):
+    entry1 = TranslationMemoryEntry.objects.create(
+        source="Hello",
+        target="Hola",
+        locale=locale_a,
+        project=project_a,
+    )
+    entry2 = TranslationMemoryEntry.objects.create(
+        source="Goodbye",
+        target="Adiós",
+        locale=locale_a,
+        project=project_b,
+    )
+    entry3 = TranslationMemoryEntry.objects.create(
+        source="Hello",
+        target="Bonjour",
+        locale=locale_b,
+        project=project_b,
+    )
+    return [entry1, entry2, entry3]
+
+
+# @pytest.fixture()
+# def private_projects():
+#     return ProjectFactory.create_batch(3, visibility=Project.Visibility.PRIVATE)
+
+
+@pytest.mark.django_db
+def test_locale(locale_a):
+    response = APIClient().get(
+        f"/api/v2/locales/{locale_a.code}/", HTTP_ACCEPT="application/json"
+    )
+    assert response.status_code == 200
+    assert isinstance(response.data, dict)
+
+
+@pytest.mark.django_db
+def test_locales_optimization(django_assert_num_queries, locales):
```

Comment:
I would drop `_optimization` from the name, since we're also testing for a successful response.

```suggestion
def test_locales(django_assert_num_queries, locales):
```